pipeline/vars/common.groovy

```diff
@@ -99,33 +87,42 @@ def runTestSuite() {
         rc = 1
         echo "Encountered an error"
     }
-
     // Forcing cleanup
     try {
         cleanup_cmd = "PYTHONUNBUFFERED=1 ${env.WORKSPACE}/.venv/bin/python"
         cleanup_cmd += " run.py --cleanup ${instanceName}"
         cleanup_cmd += " --osp-cred ${env.HOME}/osp-cred-ci-2.yaml"
         cleanup_cmd += " --log-level debug"
-
         sh(script: "${cleanup_cmd}")
     } catch(Exception ex) {
         echo "WARNING: Encountered an error during cleanup."
         echo "Please manually verify the test artifacts are removed."
     }
-
     if (rc != 0) {
         error("Test execution has failed.")
     }
-
+    return rc
 }
 
-def sendEMail(def subjectPrefix) {
+def sendEMail(def subjectPrefix,def test_results) {
     /*
         Send an email notification.
     */
+    def body = readFile(file: "pipeline/vars/emailable-report.html")
```

Comment:
I like the idea of a separate file but can it be made into a template? If not, would it make sense to have the contents in this method?

Reply:
Hi Satya,
The file has only the styling part. I thought it will too much-unwanted data in common.py
 if want to use a template then, it requires a template file to be placed in  ${JENKINS_HOME}/email-templates/.
These styles were from our email-notification template.